src/api/procedures/addSecondaryAccountsWithAuth.ts

```diff
@@ -81,6 +81,7 @@ export async function getAuthorization(
 
   return {
     permissions: {
+      // TODO: on chain just checks if is signed by primary key -> so we should check in prepareAddSecondaryKeysWithAuth if signer is the target primary key
```

Comment:
I think there is a few methods like this. Would we need to add an extra field to permissions for that?